### PR TITLE
Improve logs in daemon

### PR DESCRIPTION
**Pull request by krhubert**
PR for discussion about dameon logs.

[Issue 196](https://github.com/mesg-foundation/core/issues/196)

Below are logs in `text` and `json` formats.

```
$ ./core
INFO[0000] Starting MESG Core vX.X.X                    
INFO[0000] Server listens on /mesg/server.sock          
INFO[0000] Server listens on [::]:50052                 
INFO[0002] finished unary call with code OK              grpc.code=OK grpc.method=ExecuteTask grpc.service=api.Core grpc.start_time="2018-08-08T21:15:30+02:00" grpc.time_ms=26.072 span.kind=server system=grpc
```

``` $ ./core
{"level":"info","msg":"Starting MESG Core vX.X.X","time":"2018-08-08T21:16:37+02:00"}
{"level":"info","msg":"Server listens on [::]:50052","time":"2018-08-08T21:16:37+02:00"}
{"level":"info","msg":"Server listens on /mesg/server.sock","time":"2018-08-08T21:16:37+02:00"}
{"grpc.code":"OK","grpc.method":"ExecuteTask","grpc.service":"api.Core","grpc.start_time":"2018-08-08T21:16:43+02:00","grpc.time_ms":27.99,"level":"info","msg":"finished unary call with code OK","span.kind":"server","system":"grpc","time":"2018-08-08T21:16:43+02:00"}
```

@mesg-foundation/core Waiting for your feedback.

### CODE REVIEW

**Comment by antho1404** [comment]
My logs doesn't look like your
```
MESG Core is running
time="2018-08-09T05:24:51Z" level=info msg="Starting MESG Core dev local"
time="2018-08-09T05:24:51Z" level=info msg="Server listens on [::]:50052"
time="2018-08-09T05:24:51Z" level=info msg="Server listens on /mesg/server.sock"
```

Probably because I'm using the `./dev-core` that is running the core inside docker and then reading the logs from docker. The logs like that are not really readable

File: execution/complete.go [hunk 1]
```diff
@@ -32,9 +31,6 @@ func (execution *Execution) Complete(output string, data map[string]interface{})
 	execution.Output = output
 	execution.OutputData = data
```
[the remark this answers is not in the document]
Comment (krhubert) [reply execution/complete.go]: Yes

**Comment by antho1404** [comment]
Everything is good, I just want to fix few things before merge:
- [ ] update the go dependencies in the dockerfile `go get github.com/sirupsen/logrus && go get github.com/grpc-ecosystem/go-grpc-middleware`
- [ ] update the changelog

Also I like the colored output that is only available with TTY so maybe we should support another flag that might be the default with "colored" or something like that that would put core in a TTY mode and have the colored outputs but this is definitely not required for this PR

**Comment by krhubert** [comment]
@antho1404 done, to merge.

**Comment by antho1404** [comment]
I'm ok with this, I'm just thinking should we add the colors on this PR or another one ? the fix is quite easy `logrus.SetFormatter(&logrus.TextFormatter{ForceColors: true})`

I really think that a nice readable log for human is important but having something in plain text or json is great like that it gives a good flexibility.

Any feedbacks ? @mesg-foundation/core

**Comment by krhubert** [comment]
I'm ok to add some flag to have colors with logs. But forcing color is a bad idea.

**Comment by antho1404** [comment]
Sorry I wasn't clear enough in my proposition of the fix, I don't want to replace the text version with the colored one, the text version is really good and a must have, i was talking about another mode like described in my previous comment with "colored".

We would have the 3 following modes:
- colored (default)
- text
- json

I would prefer colored logs by default because it's more user friendly and I think developers who have the need to extract logs will be more comfortable playing with the terminal and using flags etc... and developer who are less comfortable they would prefer something more "human readable" but I would love to have other feedbacks on that

**Comment by NicolasMahe** [comment]
I agree with @antho1404 on:
> colored (default)
text
json

The CLI is meant to run locally. Even if the Core is running on a server, the CLI should handle the remote connection itself, no need for SSH. (Like Docker Machine)

**Comment by antho1404** [comment]
Ok for me except the PR for dependencies is merged so we have to update this branch and add these dependencies in the vendor

github.com/sirupsen/logrus
github.com/grpc-ecosystem/go-grpc-middleware

File: cmd/start.go [hunk 2]
```diff
@@ -18,7 +21,41 @@ var Start = &cobra.Command{
 	DisableAutoGenTag: true,
 }
 
+// logFormatValue reprenets log format flag value.
```
Comment (NicolasMahe) [inline cmd/start.go]: Typo on "reprenets"

File: cmd/start.go [hunk 3]
```diff
@@ -18,7 +21,41 @@ var Start = &cobra.Command{
...
+		return fmt.Errorf("%s is not valid log format", value)
+	}
+	viper.Set(config.LogFormat, value)
+	*v = logFormatValue(value)
+	return nil
+}
+func (v *logFormatValue) Type() string   { return "string" }
+func (v *logFormatValue) String() string { return string(*v) }
+
+// logLevelValue reprenets log level flag value.
```
Comment (NicolasMahe) [inline cmd/start.go]: Typo on "reprenets"

**Comment by krhubert** [comment]
@NicolasMahe done

**Pull request merged**